### PR TITLE
Add fix for $parent in when called in events only

### LINKED ISSUE #12: Child element needs to fire event 2 times to mutate parent data
## Issue: 
When you try to mutate parent data from a child element on an event, you need to fire it 2 times. I have also tried to pre-bind `data-last-refresh` but it does not give me any luck.
https://lh3.googleusercontent.com/5rUjGyovBt9g16iesV8NCq6U5X4uNjZqiIm_Sz295r9WCSGbpDYFTuShn2BB47wFPsoxphIUO4ZOhSTQ3xYc=w1919-h903-rw

## Source code:
```html
<div x-data="{num: 16}">
  <!-- Outside -->
  <p x-text='num'></p>
  <button @click="num = Math.random()">Random a Number (Outside)</button>
  <div x-data>
    <!-- Inside -->
    <button @click="$parent.num = Math.random()">Random a Number (Inside)</button>
  </div>
</div>
```

### PR DESCRIPTION
**Pull request by KevinBatdorf**
Fixes #12 

Demo with PR code: https://codepen.io/KevinBatdorf/pen/769e82b50f6bbf0bd4a1d4ac960f5f8f?editors=1010

Demo not working: https://codepen.io/KevinBatdorf/pen/wvGEzre?editors=1010

### CODE REVIEW

**Comment by KevinBatdorf** [comment]
Im thinking this probably should be released before the refactor.

File: src/component.js [hunk 1]
```diff
@@ -37,6 +36,17 @@ function registerComponentMagicMethod() {
...
+        // $parent is called **only** from within an event listener and not elsewhere.
+		if (parentComponent.__x && $el.__x) {
+			$el.__x.unobservedData.$nextTick(() => {
+				Object.entries($el.__x.unobservedData.$parent).forEach(value => {
+					parentComponent.__x.$data[value[0]] = value[1]
+				})
+                parentComponent.__x.updateElements(parentComponent)
+			})
+        }
+
```
Comment (SimoTod) [inline src/component.js:49]: I'm not sure we need all this code. The difference between parent and component is that in the latter you return the proxy, in the former you return the original dataset.

If you change `return data` to return `$el.$parent` it should work.

**Comment by SimoTod** [comment]
at some point we do
```js
function allowTwoWayCommunication(data, observedComponent) {
        return new Proxy(data, {
            set(object, prop, value) {
                if (!observedComponent.__x) {
                    throw 'Error communicating with observed component'
                }
                observedComponent.__x.$data[prop] = value
                observedComponent.__x.updateElements(observedComponent)
                return true
            }
        })
    }
```

I believe `observedComponent.__x.updateElements(observedComponent)` is redundant. the previous line sets a variable on using the parent proxy which is already calling updateElements.

**Review by SimoTod — CHANGES_REQUESTED** [review]
I think we can avoid adding that code. The issue is slightly different.
I've had a look at the code overall and I found a place where we call updateElements twice in a row which we should probably remove.

File: src/component.js (hunk 1 above)
Comment (KevinBatdorf) [reply src/component.js]: Looks good to me!

**Review by SimoTod — APPROVED** [review]

**Pull request merged**